torch/distributed/_tensor/experimental/dmap.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+from typing import Callable, List, Optional, Sequence, Tuple, Union
+
+import torch
+from torch.distributed._tensor import DeviceMesh, DTensor
+from torch.distributed._tensor.placement_types import Placement
+from torch.utils._pytree import tree_flatten, tree_unflatten
+
+
+PlacementType = Sequence[Placement]
+InputPlacements = Union[PlacementType, Tuple[PlacementType, ...]]
+OutputPlacements = Union[PlacementType, Tuple[PlacementType, ...]]
+
+
+def dmap(
+    func: Callable,
+    out_placements: OutputPlacements,
+    device_mesh: Optional[DeviceMesh] = None,
```

Comment:
if both `out_placements` and `in_placements` can be None, let's put device_mesh after in_placements